lib/combine_pdf/parser.rb

```diff
@@ -323,8 +323,8 @@ def _parse_
                 str << 12
               when 48..57 # octal notation for byte?
                 rep -= 48
-                rep = (rep << 3) + (str_bytes.shift-48) if str_bytes[0].between?(48, 57)
-                rep = (rep << 3) + (str_bytes.shift-48) if str_bytes[0].between?(48, 57) && (((rep << 3) + (str_bytes[0] - 48)) <= 255)
+                rep = (rep << 3) + (str_bytes.shift-48) if str_bytes[0]&.between?(48, 57)
```

Comment:
I tested `to_i` vs. `&.` and was happy with the results:

```ruby
require 'benchmark'
COUNT = 10_000_000
STR_BYTES = "1234567".bytes
counter = 0;
Benchmark.bmbm do |x|
   x.report("to_i") { COUNT.times { |i| counter +=1 if STR_BYTES[i & 7].to_i.between?(48,57) } }
   x.report("&.") { COUNT.times { |i| counter +=1 if STR_BYTES[i & 7]&.between?(48,57) } }
end
```

Nice work!

Reply:
Awesome, glad it worked out! 😄